lib/internal/modules/cjs/loader.js

```diff
@@ -248,6 +248,18 @@ function reportModuleNotFoundToWatchMode(basePath, extensions) {
   }
 }
 
+const kPaths = Symbol('kPaths');
+const lazyPathsGetter = {
+  __proto__: null,
+  enumerable: true,
+  get() { return this[kPaths] ?? Module._nodeModulePaths(this.path); },
```

Comment:
I think this could also just replace `this.paths` with a value property once computed so subsequent accesses are not invoking a getter again & the property type change is less observable from the user land.